Automattic/wp-calypso
# Woop Installer: Add checkout step

#### Changes proposed in this Pull Request

This PR adds changes to take over when the site doesn't have the proper plan.
In short, when the site doesn't have a biz plan, the `woop` feature is not active, and when it is available, the client will redirect to ~the landing page~ checkout plan page.

It's implemented in the confirm step, adding the `plan` substep.

For this purpose, I've added the `useWoopHandling()` hook that contains other data that will be used in other stuff.


https://user-images.githubusercontent.com/77539/144066535-8fdb9077-f5b4-4838-bb92-40d636f74af5.mov


Fixes https://github.com/Automattic/wp-calypso/issues/58581

#### Testing instructions



* Test with a simple site / No proper plan
* Select the site from the select-site step
* Confirm the client redirects to the ~landing~ checkout page
* Test with a site with a proper plan.
* Confirm the `plan` step is skipped
